frontend/src/components/widgets/DataFrame/hooks/useColumnSizer.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from "react"
+
+import { GridColumn, DataEditorProps } from "@glideapps/glide-data-grid"
+
+type ColumnSizerReturn = Pick<DataEditorProps, "columns" | "onColumnResize">
+
+/**
+ * Hook to manage the interactive column resizing capabilities.
+ *
+ * @param columns - The columns of the table.
+ *
+ * @returns An object containing the following properties:
+ * - columns: The updated list of columns.
+ * - onColumnResize: The callback function to be called when a column is resized.
+ */
+function useColumnSizer(columns: GridColumn[]): ColumnSizerReturn {
+  // The columns with the corresponding empty template for every type:
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Maybe because `columnSizes` and `setColumnSizes` is out of scope that this is needed but just wanted to double check if this is needed again.

Reply:
Good catch, this was indeed unnecessary 👍